src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -838,6 +874,32 @@ public virtual NodeInputData InputData
            get { return null; }
         }
 
+        [JsonIgnore]
+        public virtual NodeOutputData OutputData
+        {
+            get
+            {
+                // Determine if the output type can be determined at this time
+                // Current enum supports String, Integer, Float, Boolean, and unknown
+                // When CachedValue is null, type is set to unknown
+                // When Concrete type is dictionary or other type not expressed in enum, type is set to unknown
+                object returnObj = CachedValue?.Data?? new object();
+                var returnType = NodeOutputData.getNodeOutputTypeFromType(returnObj.GetType());
+
+                // IntialValue is returned when the Type enum does not equal unknown
+                var returnValue = (returnType != NodeOutputTypes.unknownOutput) ? returnObj.ToString() : String.Empty; 
+                
+                return new NodeOutputData()
+                {
+                    Id = this.GUID,
```

Comment:
is this getting serialized correctly? without dashes?

Reply:
I think so... No dashes.  It is the same that is used for NodeInputData